clients/ui/frontend/src/app/pages/modelCatalog/screens/RegisterCatalogModelForm.tsx

```diff
@@ -0,0 +1,243 @@
+import { Form, FormGroup, PageSection, Stack, StackItem } from '@patternfly/react-core';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import {
+  ModelLocationType,
+  RegisterCatalogModelFormData,
+  useRegisterCatalogModelData,
+} from '~/app/pages/modelRegistry/screens/RegisterModel/useRegisterModelData';
+import RegistrationCommonFormSections from '~/app/pages/modelRegistry/screens/RegisterModel/RegistrationCommonFormSections';
+import {
+  isModelNameExisting,
+  isNameValid,
+  isRegisterCatalogModelSubmitDisabled,
+  registerModel,
+} from '~/app/pages/modelRegistry/screens/RegisterModel/utils';
+import { SubmitLabel } from '~/app/pages/modelRegistry/screens/RegisterModel/const';
+import RegisterModelDetailsFormSection from '~/app/pages/modelRegistry/screens/RegisterModel/RegisterModelDetailsFormSection';
+import RegistrationFormFooter from '~/app/pages/modelRegistry/screens/RegisterModel/RegistrationFormFooter';
+import { ModelCatalogItem } from '~/app/modelCatalogTypes';
+import { ModelRegistry, ModelRegistryMetadataType } from '~/app/types';
+import { ModelRegistryContext } from '~/app/context/ModelRegistryContext';
+import useRegisteredModels from '~/app/hooks/useRegisteredModels';
+import { extractVersionTag } from '~/app/pages/modelCatalog/utils/modelCatalogUtils';
+import ModelRegistrySelector from '~/app/pages/modelRegistry/screens/ModelRegistrySelector';
+import { registeredModelUrl } from '~/app/pages/modelRegistry/screens/routeUtils';
+import { getCatalogModelDetailsRoute } from '~/app/routes/modelCatalog/catalogModelDetails';
+import {
+  catalogParamsToModelSourceProperties,
+  modelSourcePropertiesToCustomProperties,
+} from '~/concepts/modelRegistry/utils';
+
+interface RegisterCatalogModelFormProps {
+  model: ModelCatalogItem;
+  preferredModelRegistry: ModelRegistry;
+}
+
+const RegisterCatalogModelForm: React.FC<RegisterCatalogModelFormProps> = ({
+  model,
+  preferredModelRegistry,
+}) => {
+  const navigate = useNavigate();
+  const { apiState } = React.useContext(ModelRegistryContext);
+  const [registeredModels, registeredModelsLoaded] = useRegisteredModels();
+
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
+
+  const versionTag = extractVersionTag(model.tags);
+
+  const catalogParams = {
+    modelName: model.name,
+    tag: versionTag || '',
+    sourceName: model.provider || '',
+    repositoryName: model.id || '',
+  };
+  const sourceProperties = catalogParamsToModelSourceProperties(catalogParams);
+  const sourceCustomProperties = modelSourcePropertiesToCustomProperties(sourceProperties);
```

Comment:
We don't need `sourceCustomProperties` -- we are only using the `sourceProperties`.